mmv1/products/gkeonprem/VmwareCluster.yaml

```
@@ -0,0 +1,673 @@
+            required: true
+            item_type: !ruby/object:Api::Type::NestedObject
+              properties:
+                - !ruby/object:Api::Type::String
+                  name: "pool"
+                  description: |
+                    The name of the address pool.
+                  required: true
+                - !ruby/object:Api::Type::Array
+                  name: "addresses"
+                  item_type: Api::Type::String
+                  description: |
+                    The addresses that are part of this pool. Each address 
+                    must be either in the CIDR form (1.2.3.0/24) or range 
+                    form (1.2.3.1-1.2.3.5).
+                  required: true
+                - !ruby/object:Api::Type::Boolean
+                  name: "avoidBuggyIps"
+                  description: |
+                    If true, avoid using IPs ending in .0 or .255.
+                    This avoids buggy consumer devices mistakenly dropping IPv4 traffic for
+                    those special IP addresses.
+                  default_from_api: true
+                - !ruby/object:Api::Type::Boolean
+                  name: "manualAssign"
+                  description: |
+                    If true, prevent IP addresses from being automatically assigned.
+                  default_from_api: true
+  - !ruby/object:Api::Type::NestedObject
+    name: dataplaneV2
+    description: VmwareDataplaneV2Config specifies configuration for Dataplane V2.
+    default_from_api: true
+    properties:
+      - !ruby/object:Api::Type::Boolean
+        name: dataplaneV2Enabled
+        description: Enables Dataplane V2.
+      - !ruby/object:Api::Type::Boolean
+        name: windowsDataplaneV2Enabled
+        description: Enable Dataplane V2 for clusters with Windows nodes.
+      - !ruby/object:Api::Type::Boolean
+        name: advancedNetworking
+        description: Enable advanced networking which requires dataplane_v2_enabled
+          to be set true.
+  - !ruby/object:Api::Type::Boolean
+    name: vmTrackingEnabled
+    description: Enable VM tracking.
+    default_from_api: true
+  - !ruby/object:Api::Type::NestedObject
+    name: autoRepairConfig
+    description: Configuration for auto repairing.
+    default_from_api: true
+    properties:
+      - !ruby/object:Api::Type::Boolean
+        name: enabled
+        description: Whether auto repair is enabled.
+        required: true
+  - !ruby/object:Api::Type::NestedObject
+    name: authorization
+    description: RBAC policy that will be applied and managed by GKE On-Prem.
+    properties:
+      - !ruby/object:Api::Type::Array
+        name: adminUsers
+        description: |
+          Users that will be granted the cluster-admin role on the cluster, providing
+          full access to the cluster.
+        item_type: !ruby/object:Api::Type::NestedObject
+          properties:
+            - !ruby/object:Api::Type::String
+              name: username
```

```suggestion
              name: username
              required: true
```

per https://cloud.devsite.corp.google.com/anthos/clusters/docs/on-prem-api/reference/rest/v1/Authorization#clusteruser

---

Added required.